src/components/searchBar.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import './searchBar.css';
+
+const SEARCH_BAR_HEIGHT = '36px';
+const MARGIN_TOP_BOTTOM = '10px';
+export const SEARCH_HEIGHT = `${SEARCH_BAR_HEIGHT} + 2 * ${MARGIN_TOP_BOTTOM}`;
+
+interface ISearchBarProps {
+  placeholder: string;
+}
+
+interface ISearchBarState {
+  value: string;
+}
+
+class SearchBar extends React.Component<ISearchBarProps, ISearchBarState> {
+  constructor(props: ISearchBarProps) {
+    super(props);
+    const savedValue = localStorage.getItem('searchBarValue');
+    this.state = { value: savedValue || '' };
+  }
+
+  handleChange: React.ChangeEventHandler<HTMLInputElement> = (event) => {
+    this.setState({ value: event.target.value });
+  };
+
+  handleClearClick: React.MouseEventHandler<HTMLButtonElement> = () => {
+    this.setState({ value: '' });
+  };
+
+  componentWillUnmount = () => {
+    localStorage.setItem('searchBarValue', this.state.value);
```

Comment:
I am not sure if it will work if I reload the page, most likely it will not be saved. You can use the second argument of setState, it will be triggered after state is changed